nerfstudio/models/gaussian_splatting.py

```diff
@@ -686,26 +743,8 @@ def get_loss_dict(self, outputs, batch, metrics_dict=None) -> Dict[str, torch.Te
             gt_img = batch["image"]
         Ll1 = torch.abs(gt_img - outputs["rgb"]).mean()
         simloss = 1 - self.ssim(gt_img.permute(2, 0, 1)[None, ...], outputs["rgb"].permute(2, 0, 1)[None, ...])
-        if self.step % 10 == 0:
-            # Before, we made split sh and colors onto different optimizer, with shs having a low learning rate
-            # This is slow, instead we apply a regularization every few steps
-            sh_reg = self.colors_all[:, 1:, :].norm(dim=1).mean()
-            scale_exp = torch.exp(self.scales)
-            scale_reg = (
```

Comment:
This PR removes the scale regularization (taken from PhysGauss paper), I found it's useful for reducing the number of huge spikey gaussians, did you find it impacts quality? It's ok if we need to remove it/make it optional, but if so the parameter `max_gauss_ratio` should be refactored/removed.

Reply:
My implementation was trying to mimic an implementation that is closer enough to the original GSplat paper. I can potentially add it back as an option. Do you have examples that show its effectiveness in terms of removing huge spikey gaussians?

Reply:
I think leaving it off by default with an option makes sense, the effect is mostly visible in less well reconstructed parts of the scene, for example the edges of this rose plant: 
Without regularization:
<img width="500" alt="Screenshot 2024-01-06 at 3 28 35 PM" src="https://github.com/nerfstudio-project/nerfstudio/assets/15806475/8cdfd905-c698-4672-b567-698ca8bd32ad">
With regularization:
<img width="500" alt="Screenshot 2024-01-06 at 3 28 53 PM" src="https://github.com/nerfstudio-project/nerfstudio/assets/15806475/ee43fc1b-9c3e-4ad1-b092-c5b9cb29665f">
Less elongated gaussians also makes downstream usage easier for cropping the scene and reduces popping artifacts slightly, here's some videos:
Without:

https://github.com/nerfstudio-project/nerfstudio/assets/15806475/39682096-e778-4693-af6d-398c88bdb42e

With:

https://github.com/nerfstudio-project/nerfstudio/assets/15806475/8331c341-d14c-4e07-9b11-d51b48713214